src/renderer/store/selectors/contacts.ts

```diff
@@ -208,11 +208,12 @@ const usersCertificateMapping = createSelector(
       if (current !== null && current) {
         parsedCerficated = extractPubKeyString(current)
         certObject = loadCertificate(current)
-        nickname = certObject.subject.typesAndValues[0].value.valueBlock.value
-        onionAddress = certObject.subject.typesAndValues[1].value.valueBlock.value
-        peerId = certObject.subject.typesAndValues[2].value.valueBlock.value
-        const dmPubKeyArray = certObject.subject.typesAndValues[3]?.value.valueBlock.valueHex
-        dmPubKey = arrayBufferToHexString(dmPubKeyArray)
+        if (certObject.subject.typesAndValues.length === 4) {
```

Comment:
Instead of checking length of attributes just check that you have all the values